apps/studio/pages/project/[ref]/logs/explorer/index.tsx

```diff
@@ -108,22 +122,37 @@ export const LogsExplorerPage: NextPageWithLayout = () => {
       enabled: false,
     }
   )
-
-  useEffect(() => {
-    if (warehouseError) {
-      toast.error(warehouseError.message)
-    }
-  }, [warehouseError])
-
+  const results = sourceType === 'warehouse' ? warehouseResults?.result : logData
   const isLoading = logsLoading || warehouseFetching
 
-  const [recentLogs, setRecentLogs] = useLocalStorage<LogSqlSnippets.Content[]>(
-    `project-content-${projectRef}-recent-log-sql`,
-    []
-  )
-
   const { data: warehouseCollections } = useWarehouseCollectionsQuery({ projectRef })
 
+  const { isLoading: isSubmitting, mutate: createContent } = useContentInsertMutation({
+    onError: (e) => {
+      const error = e as { message: string }
+      console.error(error)
+      setSaveModalOpen(false)
+      toast.error(`Failed to save query: ${error.message}`)
+    },
+    onSuccess: (values) => {
+      setSaveModalOpen(false)
+      toast.success(`Saved "${values.name}" log query`)
+    },
+  })
```

Comment:
Nit: do we even need insert here if we can just use upsert? 😅

Reply:
okay pushed up to swap to use PUT here!